snap/local/hooks/cmd/configure/configure.go

```diff
@@ -86,22 +157,165 @@ func serviceType(name string) int {
+	for _, s := range getEdgeXServices() {
+		if err := cli.Stop(s, false); err != nil {
+			return err
+		}
+	}
+
+	// stop Kuiper-related services
+	// TODO - kuiper will be stopped, but not restarted because
+	// additional re-configuration may be needed.
+	for _, s := range getKuiperServices() {
+		if err := cli.Stop(s, false); err != nil {
+			return err
+		}
+	}
+
+	// stop redis
+	if err := cli.Stop("redis", false); err != nil {
+		return err
+	}
+
+	// stop consul
+	if err := cli.Stop("consul", false); err != nil {
+		return err
+	}
+
+	// - clear redis password
+	if err := clearRedisConf(); err != nil {
+		return err
+	}
+
+	// - clear consul ACLs
+	if err := rmConsulAclFile(); err != nil {
+		return err
+	}
+
+	// - start required services
+	for _, s := range getRequiredServices() {
+		if err := cli.Start(s, false); err != nil {
+			return err
+		}
+	}
+
+	// Now check config status of the optional EdgeX
+	// services and restart where necessary
+	for _, s := range getEdgeXServices() {
+		status, err := cli.Config(s)
+		if err != nil {
+			return err
+		}
+
+		// walk thru remaining edgex services
+		// if status is ON, start
+		// if status isn't set, if the service is
+		// part of the enabledServices (i.e. services
+		// always started), then also start it
+		switch status {
+		case "":
+			// TODO: switch to check enabledServices list
+			// when code is re-factored to remove the
+			// status="install" logic
+			if !strings.HasPrefix(s, "core-") {
+				break
+			}
+			fallthrough
+		case ON:
+			if err := cli.Start(s, false); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
 // handleAllServices iterates through all of the services in the
 // edgexfoundry snap and:
 //
-// - queries the config option associated with the service state (on|off)
+// - queries the config option associated with the service state (on|off|'')
 // - queries the environment configuration for the service (env.<service-name>)
 //   - if env configuration for the service exists, use it to write a
 //     service-specific .env file to the service config dir in $SNAP_DATA
-// - start/stop any tightly couple services (e.g. if the secret-store
-//   is disabled, the proxy also has to come down) if required
-// - start/stop the service itself if required
+// - if deferStartup == true, continue to the next service
+// - otherwise handle runtime state changes
+//   - start/stop any tightly coupled services (e.g. if the secret-store
+//     is disabled, the proxy also has to come down) if required
+//   - start/stop the service itself if required
+//
 //
 // NOTE - at this time, this function does *not* restart a service based
 // on env configuration changes. If changes are made after a service has
 // been started, the service must be restarted manually.
 //
-func handleAllServices() error {
+func handleAllServices(deferStartup bool) error {
+	var serviceList []string
+	secretStoreActive := true
```

Comment:
a couple of lines below, in line 324 we have

```
		status, err := cli.Config(s)
		if err != nil {
			return err
		}
```
If the user doesn't use lower case then we fail when processing the status:

```
sudo snap set edgexfoundry device-virtual=ON
error: cannot perform the following tasks:
- Run configure hook of "edgexfoundry" snap (run hook "configure": exit status 1)
```

Should we change this to be case-agnostic? ( by adding something like the following) 


```
		status, err := cli.Config(s)
		if err != nil {
			return err
		}
		status, err := strings.ToLower(status)
```